### PR TITLE
IGA toolset

### PR DESCRIPTION
**Pull request by jzwar**
# Overview
Provide tools and functions that can help prototype IGA applications. These consist of mappings of basis function derivatives into the physical domain. E.g., with derivative, the spline functions are differentiated with respect to the reference coordinates. In order to get the derivatives with respect to the physical coordinates, the following mappings are applied:
```math
\frac{\partial f}{\partial x_i} = \frac{\partial f}{\partial \xi_j} \frac{\partial \xi_j}{\partial x_i}
```
Second order derivatives are a little more complicated, I will provide technotes in the documentation.

## Addressed issues
*  Features for first release

## Showcase
A short / one-liner example to highlight the (new) feature
```
import splinepy as sp
# Define the Geometry
geometry = sp.BSpline(
    degrees=[1, 1],
    control_points=[
        [0.5, 0.0],
        [0.5, 0.5],
        [0.0, 0.5],
        [1.0, 0.0],
        [1.0, 1.0],
        [0.0, 1.0],
    ],
    knot_vectors=[[0, 0, 0.5, 1, 1], [0, 0, 1, 1]],
)

# Define the solution field
solution_field = sp.BSpline(
    degrees=geometry.degrees,
    control_points=np.ones((geometry.control_points.shape[0], 1)),
    knot_vectors=geometry.knot_vectors,
)


# Use refinement
solution_field.elevate_degrees([0, 1, 0, 1])

# Retrieve Hessians
evaluation_points = solution_field.greville_abscissae
mapper = solution_field.geometry_mapper(geometry=geometry)
bf_hessian, support = mapper.basis_hessian_and_support(evaluation_points)
```
Also more efficient functions that compute more than one value at the same time, avoiding duplicated computations.

## Checklists
* [x] Documentations are up-to-date.
* [x] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

File: splinepy/spline.py
Comment (j042) [inline splinepy/spline.py:577]: let's move this to a separate module? maybe time to bring `helpme` in real life.

**Review by j042 — COMMENTED** [review]
wow so cool! let's organise modules a bit before we merge

File: docs/source/conf.py
Comment (j042) [inline docs/source/conf.py:47]:
```suggestion
```
you don't need this part

Comment (jzwar) [reply docs/source/conf.py:47]: I already deleted some parts needs force pushing

**Comment by jzwar** [comment]
unit tests still missing, documentation up to date

**Review by j042 — COMMENTED** [review]
Nice! can you make a separate PR with docs layout changes?

**Comment by j042** [comment]
can you maybe use #130 here? I fear if we don't integrate that now, we probably will never update it

**Comment by jzwar** [comment]
> can you maybe use #130 here? I fear if we don't integrate that now, we probably will never update it

Waiting for #130 to merge.

**Review by j042 — APPROVED** [review]
impressive

File: splinepy/helpme/mapper.py
Comment (jzwar) [reply splinepy/helpme/mapper.py:183]: yes

**Pull request merged**
